test/e2e/quick_start_test.go

```diff
@@ -40,6 +40,7 @@ var _ = Describe("When following the Cluster API quick-start", func() {
 			InfrastructureProvider: ptr.To("docker"),
 			PostMachinesProvisioned: func(proxy framework.ClusterProxy, namespace, clusterName string) {
 				// This check ensures that owner references are resilient - i.e. correctly re-reconciled - when removed.
```

Comment:
Q: Do we have to / want to add the other diffs from main as well? (logs in the other test & usage of `CoreFinalizersAssertionWithClassyClusters`)

Reply:
If there are no relevant flakes to address, I would prefer to keep the scope as small as possible, this is why I backported only: 

- better logging
- do not use global expect inside Eventually

But Frankly speaking, if there are no flakes, I think we should avoid the PR entirely

Reply:
At least nothing to find on [triage](https://storage.googleapis.com/k8s-triage/index.html?text=finalizer%7Cowner&job=.*-cluster-api-.*&xjob=.*-provider-.*)

Reply:
@sbueringer are you ok to defer this backport?

Reply:
I think the expect change would be still good to have (no strong opinions about the logs)

We'll have the e2e tests on this branch for another 7 months

Reply:
I was mostly having a hard time figuring out why we cherry-pick some parts and not others. So I wasn't sure what I was reviewing for